sbi/inference/trainers/base.py

```diff
@@ -306,6 +320,193 @@ def train(
+                distribution, see FAQ for details on how to use custom distributions.
+            device: torch device on which to train the neural net and on which to
+                perform all posterior operations, e.g. gpu or cpu.
+            **kwargs: Additional method-specific parameters. Supported keys:
+
+        Returns:
+            NeuralPosterior object.
+        """
+
+        if sample_with == "direct":
+            posterior_estimator = estimator
+            assert isinstance(posterior_estimator, ConditionalDensityEstimator), (
+                f"Expected posterior_estimator to be an instance of "
+                " ConditionalDensityEstimator, "
+                f"but got {type(posterior_estimator).__name__} instead."
+            )
+            return DirectPosterior(
+                posterior_estimator=posterior_estimator,
+                prior=prior,
+                device=device,
+                **(kwargs.get("direct_sampling_parameters") or {}),
+            )
+        elif sample_with in ("sde", "ode"):
+            vector_field_estimator = estimator
+            assert isinstance(
+                vector_field_estimator, ConditionalVectorFieldEstimator
+            ), (
+                f"Expected vector_field_estimator to be an instance of "
+                " ConditionalVectorFieldEstimator, "
+                f"but got {type(vector_field_estimator).__name__} instead."
+            )
+            return VectorFieldPosterior(
+                vector_field_estimator,
+                prior,
+                device=device,
+                sample_with=sample_with,
+                **kwargs,
+            )
+
+        # Posteriors requiring potential_fn and theta_transform
+        potential_fn, theta_transform = self._get_potential_function(prior, estimator)
+        if sample_with == "mcmc":
+            return MCMCPosterior(
+                potential_fn=potential_fn,
+                theta_transform=theta_transform,
+                proposal=prior,
+                method=kwargs.get("mcmc_method", "slice_np_vectorized"),
+                device=device,
+                **(kwargs.get("mcmc_parameters") or {}),
+            )
+        elif sample_with == "rejection":
+            return RejectionPosterior(
+                potential_fn=potential_fn,
+                proposal=prior,
+                device=device,
+                **(kwargs.get("rejection_sampling_parameters") or {}),
+            )
+        elif sample_with == "vi":
+            return VIPosterior(
+                potential_fn=potential_fn,
+                theta_transform=theta_transform,
+                prior=prior,
+                vi_method=kwargs.get("vi_method", "rKL"),
+                device=device,
+                **(kwargs.get("vi_parameters") or {}),
+            )
+        elif sample_with == "importance":
+            return ImportanceSamplingPosterior(
```

Comment:
I prefer one return statement at the end of a method but I know this is subjective.